src/reducers/entities/posts.js

```diff
@@ -90,6 +90,35 @@ function handleReceivedPosts(posts = {}, postsInChannel = {}, action) {
     return {posts: nextPosts, postsInChannel: nextPostsForChannel};
 }
 
+function handlePostsFromSearch(posts = {}, postsInChannel = {}, action) {
+    const newPosts = action.data.posts;
+    let info = {posts, postsInChannel};
+    const postsForChannel = new Map();
+
+    for (const id in newPosts) {
```

Comment:
This could be `for (const id of Object.keys(newPosts)) {` to avoid the `newPosts.hasOwnProperty` in the next line